### PR TITLE
Fast Foward Gym Solo

### LINKED ISSUE #46: Motor Encoder Observation Has NaN Observation Space
Seems that the motor encoder observation is trying to get information from PyBullet and it's not getting back real numbers. This actually needs to be be fixed relatively soon as this breaks the entire normalization behavior of the observation factory otherwise.

With that being said, `stable-baselines` has an automatically-normalizing environment wrapper, so I'm going to be using that for the time being. However, that performs the normalization via rms standardization, which is just a hack when we should be using min-max normalization.

### LINKED ISSUE #54: VRAM Memory Leak in env.reset()
Currently the environment is reset by removing all of the bodies in the simulation (via `removeBody()`) and reloading all of the bodies again.

Running it over and over again causes the bodies to gradually disappear--leading me to believe that there is a VRAM memory leak Pybullet's implementation.

Instead, I basically copy and pasted all of the env initiation code and plopped it right after a `env.resetSimulation()`. This logic should be encompassed in the `Solo8BaseEnv`, so it should be moved there.

### PR DESCRIPTION
**Pull request by agupta231**
Kinda of a huge PR, sorry for the bulk. tl;dr:

- `gym_solo/assets` is now a submodule to https://github.com/WPI-MMR/assets. We can use that repo for issue tracking and everything will get updated here as well
- Added our Solo8 model & examples
- Fixed Pybullet VRAM memory leak (Closes #54)
- Fixed all normalization issues (Closes #46)

I'll also add comments in the PR as well to help explain some of my logic and reasoning.

### CODE REVIEW

File: gym_solo/core/rewards.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def compute(self) -> float:
 
 
 def tolerance(x: float, bounds: Tuple[float, float] = (0., 0.), 
-              margin: float = 0., margin_value: float = 1e-6):
+              margin: float = 0., margin_value: float = .1):
```
Comment (agupta231) [inline gym_solo/core/rewards.py:377]: Experimentally found-- `1e-6` was too small. Keeping it a magic number rn cause I'm writing code to phase this function out regardless.

File: gym_solo/envs/solo8v2vanilla.py [hunk 2]
```diff
@@ -117,7 +107,18 @@ def reset(self, init_call: bool = False) -> solo_types.obs:
...
+
+    if self.config.dt:
+      self.client.setPhysicsEngineParameter(fixedTimeStep=self.config.dt, 
+                                            numSubSteps=1)
+    else:
+      self.client.setRealTimeSimulation(1)
+
+    self.client_configuration()
+
+    self.plane = self.client.loadURDF('plane.urdf')
```
Comment (agupta231) [inline gym_solo/envs/solo8v2vanilla.py:121]: Pybullet has a memory leak when you delete and reload bodies. Wasn't a problem one off, but it came up during training. This fully resets the simulation and seems to have fixed that bug.

**Review by agupta231 — COMMENTED** [review]
Just a couple of explainatory comments. As always, lmk if there are any questions on anything :)

File: examples/solo8_mmr/realtime.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+              
+          Simulation Active.
+          
+          Exit with ^C.
+          =============================================
+          """)
+
+    while True:
+      pos = float(input('Which position do you want to set all the joints to?: '))
+      if pos == 69.:
```
Comment (mahajanrevant) [inline examples/solo8_mmr/realtime.py:35]: What is up with this xD?

**Review by mahajanrevant — APPROVED** [review]
I just have a question but its minor so approving it

**Comment by mahajanrevant** [comment]
Why do we have solo8_mmr now?

File: examples/solo8_mmr/realtime.py (hunk 3 above)
Comment (agupta231) [reply examples/solo8_mmr/realtime.py:35]: oh that's the value to reset the robot lol. I'll open up another PR and get rid of it lmfao

**Comment by agupta231** [comment]
We realistically only have 2 versions: ODRI's Solo and ours. Since we aren't really tracking revisions, I figured it's probably easier just to make a unified name rather than `solo8_mmr_v2.3.4`

**Pull request merged**